tango/executor.py

```diff
@@ -41,9 +41,19 @@ def execute_step_graph(self, step_graph: StepGraph) -> str:
             for package_name in self.include_package:
                 import_extra_module(package_name)
 
-        run_name = self.workspace.register_run(
-            step for step in step_graph.values() if step.cache_results
-        )
+        if run_name is None:
+            run_name = self.workspace.register_run(
+                step for step in step_graph.values() if step.cache_results
+            )
+        else:
+            try:
+                self.workspace.registered_run(run_name)
+            except KeyError:
+                self.workspace.register_run(
+                    (step for step in step_graph.values() if step.cache_results), name=run_name
+                )
```

Comment:
If the run with that name already exists, this will do nothing? That won't work. The workspace has to know about the target steps that are in the run. If we have another run with the same steps, it'll need to be updated.

There is also a risk that you start two runs with the same name but different steps, which would write over each other. That's why runs are write-once so far. I could see an argument for making runs in a workspace more mutable, but then we have to be ready for concurrent modification of runs.

I think it would be easier if we number runs with duplicate names. I.e., you end up with `flying-weasel`, `flying-weasel-2`, `flying-weasel-3`, etc.

Reply:
Hmm yeup, good point. I did this because I need to know the run directory before executing the step graph in order to initialize logging. I'll have to rethink this.

Reply:
Fixed in e21af7f